syntax = "proto3";
//option cc_generic_services=false;
package hello; 
message helloworld 
{ 
   int32     id = 1;  // ID 
   string    str = 2;  // str 
   int32     opt = 3;  //optional field 
}

message MyRequest {
    string id = 1;
    string text = 2;
    int32 times = 3;
}

message MyResponse {
    string id = 1;
    string text = 2;
    bool result = 3;
}

service MyService {
    rpc Fool (MyRequest) returns (MyResponse); 
}
